api/v1beta1/nutanixmachine_types.go

```diff
@@ -47,7 +47,30 @@ const (
 	NutanixMachineBootstrapRefKindImage = "Image"
 )
 
+// NutanixImageLookup defines how to fetch images for the cluster
+// using the fields combined.
+type NutanixImageLookup struct {
+	// ImageLookupFormat is the naming format to look up the image for this
+	// machine It will be ignored if an explicit image is set. Supports
+	// substitutions for {{.BaseOS}} and {{.K8sVersion}} with the base OS and
+	// kubernetes version, respectively. The BaseOS will be the value in
+	// ImageLookupBaseOS and the kubernetes version as
+	// defined by the packages produced by kubernetes/release without v as a
+	// prefix: 1.13.0, 1.12.5-mybuild.1, or 1.17.3. For example, the default
+	// image format of {{.BaseOS}}-?{{.K8sVersion}}-* and ImageLookupBaseOS as "rhel-8.10" will end up
+	// searching for images that match the pattern nkp-rhel-8.10-1.30.5-* for a
+	// Machine that is targeting kubernetes v1.30.5. See
+	// also: https://golang.org/pkg/text/template/
+	// +kubebuilder:default:="{{.BaseOS}}-kube-v{{.K8sVersion}}.*"
+	Format *string `json:"format,omitempty"`
+	// ImageLookupBaseOS is the name of the base operating system to use for
+	// image lookup.
+	// +kubebuilder:validation:Required
+	BaseOS *string `json:"baseOS"`
```

Comment:
Since the field is required, I think we don't want it to be a pointer.

Reply:
The validation becomes tricky here. With a pointer we are able to do validation on it not being nil, because it's a required value

```
// +kubebuilder:validation:MinLength
``` 

we can try that with a string.

I am picking up some pointers from this [blog post](https://ahmet.im/blog/crd-generation-pitfalls/index.html#explicit-required-or-optional-on-every-field) but i would like to hear your suggestions too.

Reply:
Changed the validation can you please take a look?